keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -247,11 +249,34 @@ class SqlArtifactRepository(
           .getValues(ARTIFACT_VERSIONS.VERSION)
       }
         .sortedWith(artifact.versioningStrategy.comparator)
+        .also { versions ->
+          return if (artifact is DockerArtifact) {
+            filterDockerVersions(artifact, versions)
+          } else {
+            versions
```

Comment:
On line 247, there's some debian artifact filtering that happens in the query side so there's no debian filtering necessary here. However, on line 758, there is debian artifact filtering logic. Might want to modify this code so it filters debian artifacts using the same logic as on 758, and then you could pull that out into a function.

Reply:
Good spot. I think I'd rather move the filtering into the sql query that populates line 758 if possible, so we only fetch the data we need.. looking into it.

Reply:
Done. Didn't pull it out into a function, just copied the same sql filtering. That's... not excellent, but I think it's fine. I'm not quite sure how I would extract out that apply statement.